assets/src/scripts/editoria11y-tinymce.js

```diff
@@ -0,0 +1,298 @@
+/* global Ed11y */
+( function () {
+	const POLL_INTERVAL = 200;
+	const FULL_CHECK_INTERVAL = 12000;
+	const RECHECK_DELAY = 600;
+	const initializedEditors = new WeakSet();
+	const EXPORT_SHIM = 'try{if(typeof Ed11y!=="undefined" && !window.Ed11y){window.Ed11y=Ed11y;}}catch(e){}';
+
+	/**
+	 * Run callback when DOM is ready.
+	 *
+	 * @param {Function} cb
+	 */
+	function whenDomReady( cb ) {
+		if ( document.readyState !== 'loading' ) {
+			cb();
+		} else {
+			document.addEventListener( 'DOMContentLoaded', cb );
+		}
+	}
+
+	/**
+	 * Wait until TinyMCE editors are present.
+	 *
+	 * @param {Function} cb
+	 */
+	function waitForTinyMceEditors( cb ) {
+		if ( window.tinymce && window.tinymce.editors.length ) {
+			cb();
+			return;
+		}
+		setTimeout( function () {
+			waitForTinyMceEditors( cb );
+		}, POLL_INTERVAL );
+	}
+
+	/**
+	 * Debounce a function.
+	 *
+	 * @param {Function} fn
+	 * @param {number} ms
+	 * @returns {Function}
+	 */
+	function debounce( fn, ms ) {
+		let timer = null;
+		return function debounced( ...innerArgs ) {
+			clearTimeout( timer );
+			timer = setTimeout( function () {
+				fn.apply( null, innerArgs );
+			}, ms );
+		};
+	}
+
+	/**
+	 * Build Ed11y options tailored for TinyMCE iframe.
+	 *
+	 * @returns {object}
+	 */
+	function buildEd11yOptions() {
+		const base = ( window.ed11yVars && window.ed11yVars.options ) ? JSON.parse( JSON.stringify( window.ed11yVars.options ) ) : {};
+		base.checkRoots = 'body#tinymce';
+		base.ignoreElements = ( base.ignoreElements || '' ) + ', #wpadminbar *';
+		base.liveCheck = base.liveCheck || 'all';
+		base.showResults = true;
+		base.buttonZIndex = 99999;
+		base.inlineAlerts = false;
+		base.customTests = base.customTests || 0;
+		base.preventCheckingIfPresent = '';
+		return base;
+	}
+
+	/**
+	 * Ensure Ed11y script is loaded inside iframe and constructor exported.
+	 *
+	 * @param {Window} iframeWin
+	 * @param {Function} done
+	 */
+	function ensureEd11yInIframe( iframeWin, done ) {
+		if ( iframeWin.Ed11y ) {
+			return done();
+		}
+		const parentScript = document.querySelector( 'script[src*="editoria11y.min.js"]' );
+		if ( ! parentScript ) {
+			return done();
+		}
+		if ( iframeWin.document.querySelector( 'script[src*="editoria11y.min.js"]' ) ) {
+			let loops = 0;
+			( function waitCtor() {
+				try {
```

Comment:
Inject a tiny export shim inside the iframe to allow it to get the `Ed11y` object definition within it